.github/workflows/ci.yml

```diff
@@ -147,97 +147,52 @@ jobs:
-    name: Linux
+  build-wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    needs: pre-deploy
     strategy:
       matrix:
-        pyver: [cp36-cp36m, cp37-cp37m, cp38-cp38, cp39-cp39]
-        arch: [x86_64, aarch64, i686, ppc64le, s390x]
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
       fail-fast: false
-    runs-on: ubuntu-latest
-    env:
-      py: /opt/python/${{ matrix.pyver }}/bin/python
-      img: quay.io/pypa/manylinux2014_${{ matrix.arch }}
-    needs: pre-deploy
     steps:
     - name: Checkout
       uses: actions/checkout@v2.3.4
     - name: Set up QEMU
-      id: qemu
+      if: runner.os == 'Linux'
       uses: docker/setup-qemu-action@v1
-    - name: Available platforms
-      run: echo ${{ steps.qemu.outputs.platforms }}
-    - name: Install tools
-      run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} ${{ env.py }} -m pip install -U setuptools wheel
-    - name: Make wheel
-      run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} ${{ env.py }} setup.py bdist_wheel
-    - name: Repair wheel wheel
-      run: |
-        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
-          ${{ env.img }} auditwheel repair dist/*.whl --wheel-dir wheelhouse/
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v2
       with:
-        name: dist
-        path: wheelhouse/*
-    
-  build-binary:
-    name: Binary wheels
-    strategy:
-      matrix:
-        pyver: [3.6, 3.7, 3.8, 3.9]
-        os: [macos, windows]
-        arch: [x86, x64]
-        exclude:
-        - os: macos
-          arch: x86
-      fail-fast: false
-    runs-on: ${{ matrix.os }}-latest
-    needs: pre-deploy
-    steps:
-    - name: Checkout
-      uses: actions/checkout@v2.3.4
-    - name: Setup Python 3.8
-      uses: actions/setup-python@v2.2.2
-      with:
-        python-version: ${{ matrix.pyver }}
-        architecture: ${{ matrix.arch }}
-    - name: Install dependencies
-      run: |
-        python -m pip install -U setuptools wheel          
-    - name: Make wheel
-      run:
-        python setup.py bdist_wheel
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+        platforms: all
+      id: qemu
+    - name: Available platforms
+      if: runner.os == 'Linux'
+      run: echo "${{ steps.qemu.outputs.platforms }}"
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.2.0a1
+      env:
+        # configure cibuildwheel to build native archs ('auto'), and some
+        # emulated ones
+        CIBW_ARCHS_LINUX: auto aarch64 ppc64le s390x
+        CIBW_ARCHS_MACOS: x86_64 arm64 universal2
+    - uses: actions/upload-artifact@v2
       with:
         name: dist
-        path: dist
+        path: ./wheelhouse/*.whl
 
   deploy:
     name: Deploy
-    needs: [build-linux, build-binary, build-tarball]
+    environment: release
```

Comment:
@asvetlov with this, you'll need to approve the job for it to run. Try it out before you merge. Start the release flow as usual (by pushing a tag) and you should get a notification to approve when the workflow gets to this place. I've also set a 15-minute cooldown period so that accidental releases could be canceled if anything arises.

Reply:
Thanks, I'm willing to try cooldown+approval (not sure if these steps are necessary, maybe because of the lack of experience).
Per-env secrets are definitely the thing!

Reply:
Yep, it's an extra level of security they've added and I've only started using it recently.